leafletwidget/leafletwidget.js

```diff
@@ -351,45 +346,55 @@ require(["widgets/js/widget", "leaflet", "leaflet_draw"], function(WidgetManager
         },
 
     });
-    WidgetManager.register_widget_view('LeafletDrawControlView', LeafletDrawControlView);
+    manager.WidgetManager.register_widget_view('LeafletDrawControlView', LeafletDrawControlView);
 
 
-    var LeafletMapView = IPython.DOMWidgetView.extend({
+    var LeafletMapView = widget.DOMWidgetView.extend({
         
-        initialize: function (options) {
-            LeafletMapView.__super__.initialize.apply(this, arguments);
-            this.rendered = false;
+        initialize: function() {
+            console.log('MapView initialized');
+            LeafletMapView.__super__.initialize.apply(this,arguments);
+            this.rendered=false;
         },
     
         // Layer management
         update_layers: function (old_list, new_list) {
-            this.do_diff(
-                old_list,
-                new_list, 
-                $.proxy(this.remove_layer_model, this),
-                $.proxy(this.add_layer_model, this)
-            );
+	    console.log('inside updating layers');
+	    console.log(old_list);
+	    console.log(new_list);
+            //_.difference(old_list,new_list).map(function(c) {this.remove_layer_model(c);});
+	    console.log('the difference');
+	    console.log(_.difference(new_list,old_list));
+	    console.log('adding new layers');
+            //_.difference(new_list,old_list).map(function(c) {$el.addClass(c);});
+	    this.add_layer_model(new_list[0]);
         },
```

Comment:
Ok, here's where I'm struggling. I saw a conversation about deprecating do_diff and adding promises, but I'm not sure how to update this critical bit of code.